### ISSUE TITLE
Build failing

### ISSUE THREAD

**Issue by maujim** [ros-melodic-arch/ros-melodic-image-proc#1]
Trying to install ros-melodic-desktop-full. This build is failing for me with following error message:

```Scanning dependencies of target image_proc_gencfg
[  6%] Generating dynamic reconfigure files from cfg/Resize.cfg: /home/mukund/.cache/yay/ros-melodic-image-proc/src/build/devel/include/image_proc/ResizeConfig.h /home/mukund/.cache/yay/ros-melodic-image-proc/src/build/devel/lib/python3.8/site-packages/image_proc/cfg/ResizeConfig.py
[ 13%] Generating dynamic reconfigure files from cfg/CropDecimate.cfg: /home/mukund/.cache/yay/ros-melodic-image-proc/src/build/devel/include/image_proc/CropDecimateConfig.h /home/mukund/.cache/yay/ros-melodic-image-proc/src/build/devel/lib/python3.8/site-packages/image_proc/cfg/CropDecimateConfig.py
[ 26%] Generating dynamic reconfigure files from cfg/Debayer.cfg: /home/mukund/.cache/yay/ros-melodic-image-proc/src/build/devel/include/image_proc/DebayerConfig.h /home/mukund/.cache/yay/ros-melodic-image-proc/src/build/devel/lib/python3.8/site-packages/image_proc/cfg/DebayerConfig.py
[ 26%] Generating dynamic reconfigure files from cfg/Rectify.cfg: /home/mukund/.cache/yay/ros-melodic-image-proc/src/build/devel/include/image_proc/RectifyConfig.h /home/mukund/.cache/yay/ros-melodic-image-proc/src/build/devel/lib/python3.8/site-packages/image_proc/cfg/RectifyConfig.py
Generating reconfiguration files for Resize in image_proc
Wrote header file in /home/mukund/.cache/yay/ros-melodic-image-proc/src/build/devel/include/image_proc/ResizeConfig.h
Generating reconfiguration files for Rectify in image_proc
Wrote header file in /home/mukund/.cache/yay/ros-melodic-image-proc/src/build/devel/include/image_proc/RectifyConfig.h
Generating reconfiguration files for Debayer in image_proc
Wrote header file in /home/mukund/.cache/yay/ros-melodic-image-proc/src/build/devel/include/image_proc/DebayerConfig.h
Generating reconfiguration files for CropDecimate in image_proc
Wrote header file in /home/mukund/.cache/yay/ros-melodic-image-proc/src/build/devel/include/image_proc/CropDecimateConfig.h
[ 26%] Built target image_proc_gencfg
Scanning dependencies of target image_proc
[ 33%] Building CXX object CMakeFiles/image_proc.dir/src/libimage_proc/processor.cpp.o
[ 40%] Building CXX object CMakeFiles/image_proc.dir/src/nodelets/rectify.cpp.o
[ 46%] Building CXX object CMakeFiles/image_proc.dir/src/nodelets/debayer.cpp.o
[ 53%] Building CXX object CMakeFiles/image_proc.dir/src/nodelets/resize.cpp.o
[ 60%] Building CXX object CMakeFiles/image_proc.dir/src/nodelets/crop_decimate.cpp.o
[ 66%] Building CXX object CMakeFiles/image_proc.dir/src/libimage_proc/advertisement_checker.cpp.o
[ 73%] Building CXX object CMakeFiles/image_proc.dir/src/nodelets/edge_aware.cpp.o
make[2]: *** No rule to make target '/usr/lib/libopencv_calib3d.so.4.1.2', needed by 'devel/lib/libimage_proc.so'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 80%] Building CXX object CMakeFiles/image_proc.dir/src/nodelets/crop_non_zero.cpp.o
make[1]: *** [CMakeFiles/Makefile2:506: CMakeFiles/image_proc.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
Error making: ros-melodic-image-proc
```

**Comment by acxz** [comment]
hmm, builds works fine for me. Do you have `opencv` installed? Maybe the dep on opencv3 needs to be updated to opencv version 4?

Try reinstalling all of your ros-melodic-* packages, seems like one of the packages is looking for an older `opencv` version that does not exist on your system. A rebuild will resolve the issue. Let us know the status after a clean rebuild of your `ros-melodic-* packages.

**Comment by maujim** [comment]
I have `opencv 4.2.0-1` installed. I'll try the clean rebuild and get back to you. 

Just to confirm: a clean rebuild involves delete existing packages on the machine and then reinstalling them? Or is there a faster way? I'm using `yay`

**Comment by acxz** [comment]
A clean rebuild does not mean just a reinstall. A reinstall will just delete the packages and install them again (most likely using the previous build information which is not what we want). You need to delete the existing packages build cache and then reinstall.

If you are using yay, I would suggest the following command: `yay --rebuildtree ros-melodic-image-proc`
Here is the description for it via `man yay`:
```
--rebuildtree
              When installing an AUR package rebuild and reinstall all of its AUR dependencies recursively, even
              the ones already installed. This flag allows you to easily rebuild packages against your current
              system's libraries if they have become incompatible.

```

**Comment by maujim** [comment]
Tried that with no success; got the exact same output. I suspect that the `--rebuildtree` flag isn't doing anything as it didn't rebuild the dependencies for this package. I've opened an [issue](https://github.com/Jguer/yay/issues/1141) on the `yay` repo.

There's a [similar issue](https://github.com/Jguer/yay/issues/1117) but that fix didn't work for me. I'm also on Manjaro, not vanillla Arch so not sure if that's causing any additional problems.
